chore: update doc

chuang8511 opened the pull request:
Because

- we update compogen tool
- we have two new components, email and anthropic
- we have new tasks for GoogleCloudStorage and BigQuery

This commit

- update the doc for the change

chuang8511 commented:
@GeorgeWilliamStrong cc'd @EiffelFly 
I want to quickly ask what I should do for the failure of CI process.
This PR is only updated by executing `update-component.sh` .

jvallesm commented on update-component.sh:
```diff
@@ -12,13 +12,13 @@ echo "Running go generate..."
     # echo "1:  $1"
     case $1 in
-    instill | archetypeai | huggingface | openai | stabilityai)
+    instill | archetypeai | huggingface | openai | stabilityai | anthropic)
```
Not in the scope of this PR but we could use `cut` to map the directory of the source files to the destination component.

GeorgeWilliamStrong reviewed (commented):
I am not seeing these new sections appearing in the documentation site in the deployment or when I launch locally in a dev environment. 

It looks like `content.config.tsx` file isn't being updated correctly as it doesn't contain any of the new sections?

cc'd @jvallesm @EiffelFly @orangecoloured

chuang8511 commented:
> It looks like content.config.tsx file isn't being updated correctly as it doesn't contain any of the new sections?

I am not sure what I should do here. How should I update for this point? 
Thank you! 🙇

GeorgeWilliamStrong commented on docs/component/ai/anthropic.en.mdx:
```diff
@@ -0,0 +1,70 @@
+
+| Field | Field ID | Type | Note |
+| :--- | :--- | :--- | :--- |
+| API Key (required) | `api-key` | string | Fill your Anthropic API key. To find your keys, visit the Anthropic console page. |
```
```suggestion
| API Key (required) | `api-key` | string | Fill in your Anthropic API key. To find your keys, visit the Anthropic console page. |
```

GeorgeWilliamStrong commented on docs/component/ai/anthropic.zh-CN.mdx:
```diff
@@ -0,0 +1,70 @@
+
+| Field | Field ID | Type | Note |
+| :--- | :--- | :--- | :--- |
+| API Key (required) | `api-key` | string | Fill your Anthropic API key. To find your keys, visit the Anthropic console page. |
```
```suggestion
| API Key (required) | `api-key` | string | Fill in your Anthropic API key. To find your keys, visit the Anthropic console page. |
```

GeorgeWilliamStrong commented on docs/component/ai/anthropic.en.mdx:
```diff
@@ -0,0 +1,70 @@
+| Task ID (required) | `task` | string | `TASK_TEXT_GENERATION_CHAT` |
+| Model Name (required) | `model-name` | string | The Anthropic model to be used. |
+| Prompt (required) | `prompt` | string | The prompt text |
+| System message | `system-message` | string | The system message helps set the behavior of the assistant. For example, you can modify the personality of the assistant or provide specific instructions about how it should behave throughout the conversation. By default, the model’s behavior is using a generic message as "You are a helpful assistant." |
```
```suggestion
| System message | `system-message` | string | The system message helps set the behavior of the assistant. For example, you can modify the personality of the assistant or provide specific instructions about how it should behave throughout the conversation. By default, the model’s behavior is set using a generic message as "You are a helpful assistant." |
```

GeorgeWilliamStrong commented on docs/component/ai/anthropic.zh-CN.mdx:
```diff
@@ -0,0 +1,70 @@
+| Task ID (required) | `task` | string | `TASK_TEXT_GENERATION_CHAT` |
+| Model Name (required) | `model-name` | string | The Anthropic model to be used. |
+| Prompt (required) | `prompt` | string | The prompt text |
+| System message | `system-message` | string | The system message helps set the behavior of the assistant. For example, you can modify the personality of the assistant or provide specific instructions about how it should behave throughout the conversation. By default, the model’s behavior is using a generic message as "You are a helpful assistant." |
```
```suggestion
| System message | `system-message` | string | The system message helps set the behavior of the assistant. For example, you can modify the personality of the assistant or provide specific instructions about how it should behave throughout the conversation. By default, the model’s behavior is set using a generic message as "You are a helpful assistant." |
```

GeorgeWilliamStrong commented on docs/component/ai/anthropic.zh-CN.mdx:
```diff
@@ -0,0 +1,70 @@
+| Model Name (required) | `model-name` | string | The Anthropic model to be used. |
+| Prompt (required) | `prompt` | string | The prompt text |
+| System message | `system-message` | string | The system message helps set the behavior of the assistant. For example, you can modify the personality of the assistant or provide specific instructions about how it should behave throughout the conversation. By default, the model’s behavior is using a generic message as "You are a helpful assistant." |
+| Extra Parameters | `extra-params` | object | Extra Parameters |
```
```suggestion
| Extra Parameters | `extra-params` | object | Extra parameters |
```

Please can you add a suitable description of the extra parameters. The Description field shouldn't copy the Input.

GeorgeWilliamStrong commented on docs/component/ai/anthropic.en.mdx:
```diff
@@ -0,0 +1,70 @@
+| Model Name (required) | `model-name` | string | The Anthropic model to be used. |
+| Prompt (required) | `prompt` | string | The prompt text |
+| System message | `system-message` | string | The system message helps set the behavior of the assistant. For example, you can modify the personality of the assistant or provide specific instructions about how it should behave throughout the conversation. By default, the model’s behavior is using a generic message as "You are a helpful assistant." |
+| Extra Parameters | `extra-params` | object | Extra Parameters |
```
```suggestion
| Extra Parameters | `extra-params` | object | Extra parameters |
```

Please can you add a suitable description of the extra parameters. The Description field shouldn't copy the Input.

GeorgeWilliamStrong commented on docs/component/ai/anthropic.en.mdx:
```diff
@@ -0,0 +1,70 @@
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_TEXT_GENERATION_CHAT` |
+| Model Name (required) | `model-name` | string | The Anthropic model to be used. |
```
```suggestion
| Model Name (required) | `model-name` | string | The Anthropic model to be used |
```

GeorgeWilliamStrong commented on docs/component/ai/anthropic.zh-CN.mdx:
```diff
@@ -0,0 +1,70 @@
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_TEXT_GENERATION_CHAT` |
+| Model Name (required) | `model-name` | string | The Anthropic model to be used. |
```
```suggestion
| Model Name (required) | `model-name` | string | The Anthropic model to be used |
```

GeorgeWilliamStrong commented on docs/component/anthropic.en.mdx:
Think this file shouldn't be here

GeorgeWilliamStrong commented on docs/component/anthropic.zh-CN.mdx:
Think this file shouldn't be here

GeorgeWilliamStrong commented on docs/component/application/email.en.mdx:
```diff
@@ -0,0 +1,152 @@
+- [Send Email](#send-email)
+- [Read Emails](#read-emails)
+
+You can connect different email server through Email component.
```
```suggestion
You can connect to different email servers through the Email component.
```

GeorgeWilliamStrong commented on docs/component/application/email.zh-CN.mdx:
```diff
@@ -0,0 +1,152 @@
+- [Send Email](#send-email)
+- [Read Emails](#read-emails)
+
+You can connect different email server through Email component.
```
```suggestion
You can connect to different email servers through the Email component.
```

GeorgeWilliamStrong commented on docs/component/application/email.en.mdx:
```diff
@@ -0,0 +1,152 @@
+| :--- | :--- | :--- | :--- |
+| Server Address (required) | `server-address` | string | The address of the email server |
+| Server Port (required) | `server-port` | integer | The port of the email server |
+| Email Address (required) | `email-address` | string | Email address of the user |
+| App Password (required) | `password` | string | Password of the App passwords in Gmail settings |
```
```suggestion
| Email Address (required) | `email-address` | string | The email address of the user |
| App Password (required) | `password` | string | The password of the App passwords in Gmail settings |
```

GeorgeWilliamStrong commented on docs/component/application/email.zh-CN.mdx:
```diff
@@ -0,0 +1,152 @@
+| :--- | :--- | :--- | :--- |
+| Server Address (required) | `server-address` | string | The address of the email server |
+| Server Port (required) | `server-port` | integer | The port of the email server |
+| Email Address (required) | `email-address` | string | Email address of the user |
+| App Password (required) | `password` | string | Password of the App passwords in Gmail settings |
```
```suggestion
| Email Address (required) | `email-address` | string | The email address of the user |
| App Password (required) | `password` | string | The password of the App passwords in Gmail settings |
```

GeorgeWilliamStrong commented on docs/component/application/email.en.mdx:
```diff
@@ -0,0 +1,152 @@
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_READ_EMAILS` |
+| Search | `search` | object | the search criteria for the emails |
```
```suggestion
| Search | `search` | object | The search criteria for the emails |
```

GeorgeWilliamStrong commented on docs/component/application/email.zh-CN.mdx:
```diff
@@ -0,0 +1,152 @@
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_READ_EMAILS` |
+| Search | `search` | object | the search criteria for the emails |
```
```suggestion
| Search | `search` | object | The search criteria for the emails |
```

GeorgeWilliamStrong commented on docs/component/application/email.en.mdx:
```diff
@@ -0,0 +1,152 @@
+description: "Learn about how to set up a VDP Email component https://github.com/instill-ai/instill-core"
+---
+
+The Email component is a application component that allows users to get and send email from Mail Protocol.
```
```suggestion
The Email component is an application component that allows users to get and send email from Mail Protocol.
```

GeorgeWilliamStrong commented on docs/component/application/email.zh-CN.mdx:
```diff
@@ -0,0 +1,152 @@
+description: "Learn about how to set up a VDP Email component https://github.com/instill-ai/instill-core"
+---
+
+The Email component is a application component that allows users to get and send email from Mail Protocol.
```
```suggestion
The Email component is an application component that allows users to get and send email from Mail Protocol.
```

GeorgeWilliamStrong commented on docs/component/data/bigquery.en.mdx:
```diff
@@ -9,35 +9,78 @@ The BigQuery component is a data component that allows users to insert data to B
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_READ` |
+| Filtering | `filtering` | string | The filter to be applied to the data, please start with where clause |
```
"please start with where clause" doesn't sound right. Doesn't it need to start with where clause? If so, let's state it.

chuang8511 replied:
@GeorgeWilliamStrong 
> user may need a bit more of an explanation/example or link to external documentation to supplement the filtering description?

This filter function is for the users who can use SQL command. So, I wrote the description like below. 
Please check it.

"The filter to be applied to the data with SQL syntax, which starts with WHERE clause"

GeorgeWilliamStrong commented on docs/component/data/bigquery.zh-CN.mdx:
```diff
@@ -9,35 +9,78 @@ The BigQuery component is a data component that allows users to insert data to B
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_READ` |
+| Filtering | `filtering` | string | The filter to be applied to the data, please start with where clause |
```
"please start with where clause" doesn't sound right. Doesn't it need to start with where clause? If so, let's state it.

GeorgeWilliamStrong replied:
Actually, I think the user may need a bit more of an explanation/example or link to external documentation to supplement the filtering description? What do you think?

GeorgeWilliamStrong commented on docs/component/data/googlecloudstorage.en.mdx:
```diff
@@ -9,38 +9,119 @@ The Google Cloud Storage component is a data component that allows users to uplo
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_READ_OBJECTS` |
+| Bucket Name (required) | `bucket-name` | string | Name of the bucket to be used for object storage. |
```
```suggestion
| Bucket Name (required) | `bucket-name` | string | Name of the bucket to be used for object storage |
```

GeorgeWilliamStrong commented on docs/component/data/googlecloudstorage.en.mdx:
```diff
@@ -9,38 +9,119 @@ The Google Cloud Storage component is a data component that allows users to uplo
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_CREATE_BUCKET` |
+| Bucket Name (required) | `bucket-name` | string | Name of the bucket to be used for object storage. |
```
```suggestion
| Bucket Name (required) | `bucket-name` | string | Name of the bucket to be used for object storage |
```

GeorgeWilliamStrong commented on docs/component/data/googlecloudstorage.zh-CN.mdx:
```diff
@@ -9,38 +9,119 @@ The Google Cloud Storage component is a data component that allows users to uplo
 | Input | ID | Type | Description |
 | :--- | :--- | :--- | :--- |
 | Task ID (required) | `task` | string | `TASK_UPLOAD` |
+| Bucket Name (required) | `bucket-name` | string | Name of the bucket to be used for object storage. |
```
```suggestion
| Bucket Name (required) | `bucket-name` | string | Name of the bucket to be used for object storage |
```

GeorgeWilliamStrong commented on docs/component/data/googlecloudstorage.zh-CN.mdx:
```diff
@@ -9,38 +9,119 @@ The Google Cloud Storage component is a data component that allows users to uplo
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_READ_OBJECTS` |
+| Bucket Name (required) | `bucket-name` | string | Name of the bucket to be used for object storage. |
```
```suggestion
| Bucket Name (required) | `bucket-name` | string | Name of the bucket to be used for object storage |
```

GeorgeWilliamStrong commented on docs/component/data/googlecloudstorage.zh-CN.mdx:
```diff
@@ -9,38 +9,119 @@ The Google Cloud Storage component is a data component that allows users to uplo
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_CREATE_BUCKET` |
+| Bucket Name (required) | `bucket-name` | string | Name of the bucket to be used for object storage. |
```
```suggestion
| Bucket Name (required) | `bucket-name` | string | Name of the bucket to be used for object storage |
```

GeorgeWilliamStrong commented on docs/component/data/googlecloudstorage.en.mdx:
```diff
@@ -9,38 +9,119 @@ The Google Cloud Storage component is a data component that allows users to uplo
 | Input | ID | Type | Description |
 | :--- | :--- | :--- | :--- |
 | Task ID (required) | `task` | string | `TASK_UPLOAD` |
+| Bucket Name (required) | `bucket-name` | string | Name of the bucket to be used for object storage. |
```
```suggestion
| Bucket Name (required) | `bucket-name` | string | Name of the bucket to be used for object storage |
```

GeorgeWilliamStrong commented on docs/component/data/pinecone.en.mdx:
```diff
@@ -11,27 +11,39 @@ It can carry out the following tasks:
 | Field | Field ID | Type | Note |
 | :--- | :--- | :--- | :--- |
-| API Key (required) | `api-key` | string | Fill your Pinecone AI API key. You can create a api key in Pinecone Console |
+| API Key (required) | `api-key` | string | Fill in your Pinecone AI API key. You can create a api key in Pinecone Console |
```
```suggestion
| API Key (required) | `api-key` | string | Fill in your Pinecone AI API key. You can create an api key in Pinecone Console |
```

GeorgeWilliamStrong commented on docs/component/data/pinecone.zh-CN.mdx:
```diff
@@ -11,27 +11,39 @@ It can carry out the following tasks:
 | Field | Field ID | Type | Note |
 | :--- | :--- | :--- | :--- |
-| API Key (required) | `api-key` | string | Fill your Pinecone AI API key. You can create a api key in Pinecone Console |
+| API Key (required) | `api-key` | string | Fill in your Pinecone AI API key. You can create a api key in Pinecone Console |
```
```suggestion
| API Key (required) | `api-key` | string | Fill in your Pinecone AI API key. You can create an api key in Pinecone Console |
```

GeorgeWilliamStrong reviewed (changes requested):
Looks great, just submitted some very minor details I'd like to iron out

jvallesm commented:
> It looks like content.config.tsx file isn't being updated correctly as it doesn't contain any of the new sections?

Yeah the script won't update that file, so for new pages we need to modify it manually, at least for now.

chuang8511 commented:
@GeorgeWilliamStrong 
Sync
I fixed all typos [here](https://github.com/instill-ai/component/pull/189) .

@jvallesm 
Sync
I fixed a small part, "a" & "an", for [compogen tool](https://github.com/instill-ai/component/pull/189/files#diff-fe594a06bc819872e12d6edd001e4006c8ead9268284bbad558be80a268ad707L25) .

chuang8511 commented:
Converted to draft because it needs to wait for [this PR](https://github.com/instill-ai/component/pull/189)

GeorgeWilliamStrong commented on docs/component/data/bigquery.en.mdx:
```diff
@@ -9,35 +9,78 @@ The BigQuery component is a data component that allows users to insert data to B
 | :--- | :--- | :--- | :--- |
 | Task ID (required) | `task` | string | `TASK_INSERT` |
-| Data (required) | `data` | object | The data to be inserted to BigQuery |
+| Data | `data` | object | The data to be inserted to BigQuery |
```
@chuang8511 any update

chuang8511 replied:
If there is not result, the data will be blank.

GeorgeWilliamStrong replied:
Fair enough

GeorgeWilliamStrong commented on docs/component/data/bigquery.zh-CN.mdx:
```diff
@@ -9,35 +9,78 @@ The BigQuery component is a data component that allows users to insert data to B
 | :--- | :--- | :--- | :--- |
 | Task ID (required) | `task` | string | `TASK_INSERT` |
-| Data (required) | `data` | object | The data to be inserted to BigQuery |
+| Data | `data` | object | The data to be inserted to BigQuery |
```
@chuang8511 any update

chuang8511 replied:
If there is not result, the data will be blank.

GeorgeWilliamStrong commented on docs/component/data/googlecloudstorage.en.mdx:
```diff
@@ -9,38 +9,119 @@ The Google Cloud Storage component is a data component that allows users to uplo
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_READ_OBJECTS` |
+| Bucket Name (required) | `bucket-name` | string | Name of the bucket to be used for object storage. |
+| Delimiter | `delimiter` | string | The delimiter to use when listing objects |
+| Prefix | `prefix` | string | The prefix to use when listing objects |
+| Versions | `versions` | boolean | Whether to list all versions of an object |
+| Start Offset | `start-offset` | string | The offset to start listing objects from |
+| End Offset | `end-offset` | string | The offset to stop listing objects at |
+| Include Trailing Delimiter | `include-trailing-delimiter` | boolean | Whether to include the delimiter in the object name |
+| Match Glob | `match-glob` | string | Whether to match objects using a glob pattern |
+| Include Folders As Prefixes | `include-folders-as-prefixes` | boolean | Whether to include folders as prefixes |
```
@chuang8511 we will need default values for non-required arguments

chuang8511 replied:
Will be checked in the future sprint

GeorgeWilliamStrong commented on docs/component/data/googlecloudstorage.zh-CN.mdx:
```diff
@@ -9,38 +9,119 @@ The Google Cloud Storage component is a data component that allows users to uplo
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_READ_OBJECTS` |
+| Bucket Name (required) | `bucket-name` | string | Name of the bucket to be used for object storage. |
+| Delimiter | `delimiter` | string | The delimiter to use when listing objects |
+| Prefix | `prefix` | string | The prefix to use when listing objects |
+| Versions | `versions` | boolean | Whether to list all versions of an object |
+| Start Offset | `start-offset` | string | The offset to start listing objects from |
+| End Offset | `end-offset` | string | The offset to stop listing objects at |
+| Include Trailing Delimiter | `include-trailing-delimiter` | boolean | Whether to include the delimiter in the object name |
+| Match Glob | `match-glob` | string | Whether to match objects using a glob pattern |
+| Include Folders As Prefixes | `include-folders-as-prefixes` | boolean | Whether to include folders as prefixes |
```
@chuang8511  we will need default values for non-required arguments

chuang8511 replied:
Will be checked in the future sprint

GeorgeWilliamStrong reviewed (changes requested):
@chuang8511 Need to update non-required arguments with their default values

chuang8511 commented on docs/component/data/googlecloudstorage.zh-CN.mdx:
```diff
@@ -9,38 +9,119 @@ The Google Cloud Storage component is a data component that allows users to uplo
+| Task ID (required) | `task` | string | `TASK_CREATE_BUCKET` |
+| Bucket Name (required) | `bucket-name` | string | Name of the bucket to be used for object storage. |
+| Project ID (required) | `project-id` | string | The project ID to create the bucket in |
+| Location | `location` | string | The location to create the bucket in |
```
Will be checked in the future sprint

GeorgeWilliamStrong reviewed (approved):
Approve, subject to updating default values for non-required arguments in future
